src/torchcodec/_core/Encoder.cpp

```diff
@@ -140,11 +132,50 @@ AudioEncoder::AudioEncoder(
   streamIndex_ = avStream->index;
 }
 
+AVSampleFormat AudioEncoder::findOutputSampleFormat(const AVCodec& avCodec) {
+  // Find a sample format that the encoder supports. We prefer using FLT[P],
+  // since this is the format of the input waveform. If FLTP isn't supported
+  // then we'll need to convert the AVFrame's format. Our heuristic is to encode
+  // into the format with the highest resolution.
+  if (avCodec.sample_fmts == nullptr) {
+    // Can't really validate anything in this case, best we can do is hope that
+    // FLTP is supported by the encoder. If not, FFmpeg will raise.
+    return AV_SAMPLE_FMT_FLTP;
+  }
+
+  std::vector<AVSampleFormat> preferredFormatsOrder = {
```

Comment:
We should make this a static `const` so that we only initialize it once - I think this is on the critical path for encoding performance. I think it's cleanest to make it a `const` global in just this cpp file rather than making it a static class member. 

(Since we're in C++17, we can't make it a `constexpr`, but we could potentially use `std::array` instead: https://stackoverflow.com/questions/33241909/cannot-create-constexpr-stdvector. I don't know how much that matters, but I was curious.)